addons/account/models/account_payment.py

```diff
@@ -381,6 +379,14 @@ def _compute_destination_account_id(self):
             else:
                 self.destination_account_id = self.partner_id.property_account_payable_id.id
 
+    @api.depends('move_line_ids.matched_debit_ids', 'move_line_ids.matched_credit_ids')
+    def _compute_reconciled_invoice_ids(self):
+        for record in self:
+            invoices_set = set()
+            invoices_set.update(record.move_line_ids.mapped('matched_debit_ids.debit_move_id.invoice_id.id') + record.move_line_ids.mapped('matched_credit_ids.credit_move_id.invoice_id.id'))
```

Comment:
I think you don't need to use the `set` and `browse`, you can use `|`:

`record.reconciled_invoice_ids = (record.move_line_ids.mapped('matched_debit_ids.debit_move_id.invoice_id') | record.move_line_ids.mapped('matched_credit_ids.credit_move_id.invoice_id'))`